bot/util/http.py

```diff
@@ -61,22 +62,20 @@ async def inner(*args, no_cache=False, **kwargs):
     return decorator
 
 
-# Removes the aiohttp ClientSession instance warning.
-class HTTPSession(aiohttp.ClientSession):
-    """ Abstract class for aiohttp. """
+_loop = asyncio.get_event_loop()
 
+
+class _Session(aiohttp.ClientSession):
     def __del__(self):
-        """
-        Closes the ClientSession instance
-        cleanly when the instance is deleted.
-        Useful for things like when the interpreter closes.
-        This would be perfect if discord.py had this as well. :thinking:
-        """
         if not self.closed:
-            self.close()
+            _loop.run_until_complete(self.close())
+
+
+async def generate_session():
+    return _Session()
 
 
-session = HTTPSession()
+session = _loop.run_until_complete(generate_session())
```

Comment:
This does not need to be async as initializing an object is sync.

```suggestion
session = _Session()
```

Reply:
initializing a ClientSession in a non-async function returns a DeprecationWarning

Reply:
`asyncio.get_event_loop()` was Deprecated in Python 3.10